src/body.rs

```diff
@@ -448,3 +462,33 @@ impl BufRead for Body {
         Pin::new(&mut self.reader).consume(amt)
     }
 }
+
+/// Look at first few bytes of a file to determine the mime type.
+/// This is used for various binary formats such as images and videos.
+#[cfg(feature = "async_std")]
+async fn peek_mime(file: &mut async_std::fs::File) -> io::Result<Option<Mime>> {
+    // Reading the first 8 bytes should be enough to sniff the mime type.
```

Comment:
Looking at infer's code, it seems like quite a few of the matchers go beyond 8 bytes. Is there a meaningful performance penalty to providing the first 300 bytes or something like that to infer? I can imagine a later bug where someone is like "why isn't my [jpeg2000](https://github.com/bojand/infer/blob/master/src/matchers/image.rs#L23)|[wav](https://github.com/bojand/infer/blob/master/src/matchers/audio.rs#L45)|[epub](https://github.com/bojand/infer/blob/master/src/matchers/archive.rs#L8) file being served correctly, infer supports it?" and the bugfix is to change this number

300 suggested because although happily people usually gzip their tars, the tar format seems to have quite a preamble before it gets around to saying what it is: https://github.com/bojand/infer/blob/master/src/matchers/archive.rs#L48-L55

Reply:
heh, good point. updating it to 300 bytes!